### ISSUE TITLE
playing around with unicity "groups" - can we localize infinite unicity taxonomically for Staph aureus?

### ISSUE THREAD

**Issue by ctb** [ctb/2022-sourmash-sens-spec#3]
In https://github.com/ctb/2022-sourmash-sens-spec/issues/1#issuecomment-1320896787, we saw that species `s__Staphylococcus aureus` had 8367 genomes all with infinite unicity. A basic hypothesis to test is whether all of these genomes are confused with each other, or if instead there are hashes outside of Staph that are required to reach infinite unicity.

So, I took all 8367 genomes and built a merged sketch, then ran `lca summarize` on it - in more detail, I output their identifiers to `staph_inf.csv` in column `ident, then ran:

```
sourmash sig merge --picklist staph_inf.csv:ident:ident gtdb-rs207.genomic.k31.sqldb -o staph_inf.merged.sig
gzip staph_inf.merged.sig
sourmash lca summarize --query staph_inf.merged.sig.gz --db gtdb-rs207.genomic.k31.sqldb -o staph_inf.summ.csv
```

and got the resulting output - 

|    |   count |   total_counts |   percent | superkingdom   | phylum        | class      | order               | family               | genus             | species                  |
|---:|--------:|---------------:|----------:|:---------------|:--------------|:-----------|:--------------------|:---------------------|:------------------|:-------------------------|
|  0 |    1480 |           2369 |      62.5 | d__Bacteria    | p__Firmicutes | c__Bacilli | o__Staphylococcales | f__Staphylococcaceae | g__Staphylococcus | s__Staphylococcus aureus |
|  1 |    1815 |           2369 |      76.6 | d__Bacteria    | p__Firmicutes | c__Bacilli | o__Staphylococcales | f__Staphylococcaceae | g__Staphylococcus | nan                      |
|  2 |    1861 |           2369 |      78.6 | d__Bacteria    | p__Firmicutes | c__Bacilli | o__Staphylococcales | f__Staphylococcaceae | nan               | nan                      |
|  3 |    1867 |           2369 |      78.8 | d__Bacteria    | p__Firmicutes | c__Bacilli | o__Staphylococcales | nan                  | nan               | nan                      |
|  4 |    1944 |           2369 |      82.1 | d__Bacteria    | p__Firmicutes | c__Bacilli | nan                 | nan                  | nan               | nan                      |
|  5 |    1944 |           2369 |      82.1 | d__Bacteria    | p__Firmicutes | nan        | nan                 | nan                  | nan               | nan                      |
|  6 |    2363 |           2369 |      99.7 | d__Bacteria    | nan           | nan        | nan                 | nan                  | nan               | nan                      |
|  7 |       6 |           2369 |       0.3 | nan            | nan           | nan        | nan                 | nan                  | nan               | nan                      |

and the short answer is ...nope. You get to 82% with phylum Firmicutes, but there are 6 hashes in Staph aureus genomes that are shared with Archea! (that's what the last row means)

**Comment by ctb** [comment]
I looked at GCA_019115945.1, Candidatus Borkfalkia excrementavium, ASM1911594v1, which was a singleton genome with infinite unicity - saw:

|    |   count |   total_counts |   percent | superkingdom   | phylum          | class         | order                 | family            | genus         | species                      |
|---:|--------:|---------------:|----------:|:---------------|:----------------|:--------------|:----------------------|:------------------|:--------------|:-----------------------------|
|  0 |     137 |            147 |      93.2 | d__Bacteria    | p__Firmicutes_A | c__Clostridia | o__Christensenellales | f__Borkfalkiaceae | g__Borkfalkia | s__Borkfalkia excrementavium |
|  1 |     142 |            147 |      96.6 | d__Bacteria    | p__Firmicutes_A | c__Clostridia | o__Christensenellales | f__Borkfalkiaceae | g__Borkfalkia | nan                          |
|  2 |     144 |            147 |      98   | d__Bacteria    | p__Firmicutes_A | c__Clostridia | o__Christensenellales | f__Borkfalkiaceae | nan           | nan                          |
|  3 |     145 |            147 |      98.6 | d__Bacteria    | p__Firmicutes_A | c__Clostridia | o__Christensenellales | nan               | nan           | nan                          |
|  4 |     147 |            147 |     100   | d__Bacteria    | p__Firmicutes_A | c__Clostridia | nan                   | nan               | nan           | nan                          |
|  5 |     147 |            147 |     100   | d__Bacteria    | p__Firmicutes_A | nan           | nan                   | nan               | nan           | nan                          |
|  6 |     147 |            147 |     100   | d__Bacteria    | nan             | nan           | nan                   | nan               | nan           | nan                          |

so, I think "unicity groups" as a concept are not really a useful thing, since there are hashes all over the place (taxonomically)... at any rate I have an easy way to explore them now, if we are interested.

**Comment by taylorreiter** [comment]
and charcoal didn't detect GCA_019115945.1 as contaminated 🤔

**Comment by ctb** [comment]
I don't think it is contaminated - it shares hashes with some higher taxonomies, but that's all. For gather to detect is at contaminated it would have to share multiple hashes with a specific genome in another taxonomic unit. But yeah we should look into this 😆 

I'm also wondering if we need to do something different here where we look for disagreeing hashes a la kraken rather than this. putting a pin in it.
